### DISCUSSION TITLE
Why click on thumb  does not always work ?

### DISCUSSION THREAD

**Discussion by VahromeevIlya** [nolimits4web/swiper#5725] [Q&A]
I ran into a swiper problem, clicking on the preview of a slide doesn't always work. This is evident in the official example https://swiperjs.com/demos#thumbs-gallery . Click on previews 20-30 times, and you'll see that the click does not always work. 

This problem is especially noticeable on mac os + apple mouse

If you put a click listener on the document, it works, but the internal click handler in the swiper doesn't always work.

**Comment by BjornKraft** [comment]
Adding `threshold: 5,` solved it for me.

**Reply by natitbc** [reply]
this solution also works for me, add it to thumb swiper.

![image](https://github.com/user-attachments/assets/698593d4-456f-42c5-94b7-fe5c24b72f8d)

**Comment by rabi-jollycommerce** [comment]
I am having same issue, when thumb is clicked, there is no response and main slide does not appear. I am using mac.

**Reply by VahromeevIlya** [reply]
I found a solution, check the link https://qna.habr.com/q/1152372#answer_2167314

**Reply by rabi-jollycommerce** [reply]
I didn't have to do any core change, it solved with thumb module integration.

**Comment by YouCanflyThruMostTheSun** [comment]
Is there any update on this issue? I have tried every fix I could search for yet, I still have problems with the thumbnails registering clicks.

Another odd behavior is also happening: When you get to the last 4 thumbnails, it will change the main image to the very last thumbnail and start not registering clicks.

**Reply by VahromeevIlya** [reply]
Did this solution not work for you?


```
on: {
  touchEnd: function(s,e) {
    let range = 5;
    let diff = s.touches.diff = s.isHorizontal() ? s.touches.currentX - s.touches.startX : s.touches.currentY 
                 - s.touches.startY
    if (diff < range || diff > -range) s.allowClick = true;
  }
},
```

**Reply by SebastienGicquel** [reply]
@VahromeevIlya Can you you explain your solution ? It is working but I don't understand why.

**Reply by VahromeevIlya** [reply]
> @VahromeevIlya Can you you explain your solution ? It is working but I don't understand why.

I don't know how it works either, found the answer here 
https://github.com/nolimits4web/swiper/issues/1623
And just made it easier for people to find that answer
